### PR TITLE
ENG-129851 - Button Components

### PR DESCRIPTION
**Pull request by kevinleedrum**
- Added a `dropdown` prop to `mx-button` to render the dropdown/chevron icon and make any necessary styling tweaks (i.e. Text Buttons are no longer uppercase when they have the dropdown icon).
- Swapped the `disabled` attribute for `aria-disabled` to allow for disabled buttons to be focused.
- Replaced most of the `mx-button` SCSS with tailwind classes.  The only rules left are colors, which we can tackle in a future PR.
- Implemented Icon Buttons as an additional button type in the `mx-button` component.
- Implemented Toggle Buttons with a new `mx-toggle-button` component.
- Added a Toggle Button Group (`mx-toggle-button-group`) component to easily allow toggle buttons to act as radio buttons.  I initially attempted to have toggle buttons use `<input type="radio">` under the hood, so we wouldn't need to wrap them in a "group" component, but when I tried to wire up those toggle buttons to some actual Vue state, it was a huge chore because I had to bind to the `checked` attribute and `input` event on every single toggle button (`v-model` cannot be used with custom elements).  It is much more convenient for the consumer to be able to bind to a single group component.
- Extracted the ripple effect out to separate files, and added a CSS variable for the color since it changes based on the button type.
- Added custom box-shadow values (`shadow-dp-1`, etc.) to the tailwind config since I needed one of those shadows for the chevron icon button, and I didn't want the background color from the `elevation1` class.
- Added `first-of-type` and `last-of-type` variants to tailwind, which I used to set the `border-radius` of sibling toggle buttons.
- Lots of unit test and documentation changes.

Questions:
- Should we keep tailwind's shadow utilities, or should we restrict our design system to just our elevations?  And if so, should we make them `shadow-1`, etc. instead of `shadow-dp-1` like I have now?  I only added the "dp" so our values would all be grouped together in the tailwind intellisense.
- I embedded the chevron `svg`, which I copied from the Figma file, into the `mx-button` component because I didn't want the consumer to always have to pass in an icon for that.  However, I did not bake in a "close" icon for the close icon button.  I'm still not 100% sure which icons we should hard-code vs. force the user to explicitly specify.  We'll probably run into this again when we build components with arrows, hamburgers, triple-dots, etc.
- I did not implement the icon button badge yet because I'm thinking the badge might be more versatile as a wrapper component like [this](https://material-ui.com/components/badges/).  What do you think?

https://moxiworks.atlassian.net/browse/ENG-129851

### CODE REVIEW

**Review by iamjpg — COMMENTED** [review]
This is a fantastic PR. I love the button refactor reusing the internals, adding accessibility attributes, and the addition of the toggle button.

I only had one question around the config before merging these commits - let me know what you think.

File: tailwind.config.js [hunk 1]
```diff
@@ -95,8 +111,18 @@ module.exports = {
...
+          return `.${e(`first-of-type${separator}${className}`)}:first-of-type`;
+        });
+      });
+      addVariant('last-of-type', ({ modifySelectors, separator }) => {
+        modifySelectors(({ className }) => {
+          return `.${e(`last-of-type${separator}${className}`)}:last-of-type`;
+        });
+      });
+    }),
+  ],
```
Comment (iamjpg) [inline tailwind.config.js:127]: ❤️

File: tailwind.config.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
+const plugin = require('tailwindcss/plugin');
+
 module.exports = {
-  important: '.mds',
+  important: true,
```
[the remark this answers is not in the document]
Comment (kevinleedrum) [reply tailwind.config.js:4]: Ah okay, I changed this to fix something... I don't remember what it was, but I'll take another look and see if I can fix it another way.

**Comment by iamjpg** [comment]
* Should we keep tailwind's shadow utilities, or should we restrict our design system to just our elevations? And if so, should we make them shadow-1, etc. instead of shadow-dp-1 like I have now? I only added the "dp" so our values would all be grouped together in the tailwind intellisense.

Good question: I think we should use our shadow utilities. Keeping them named inline with Tailwind makes sense to me for intellisense. Def a nice to have for our IDEs

* I embedded the chevron svg, which I copied from the Figma file, into the mx-button component because I didn't want the consumer to always have to pass in an icon for that. However, I did not bake in a "close" icon for the close icon button. I'm still not 100% sure which icons we should hard-code vs. force the user to explicitly specify. We'll probably run into this again when we build components with arrows, hamburgers, triple-dots, etc.

I imagine this will be a little fluid. I saw the SVG and think that's 100% appropriate in that case. Let's keep it case by case for the time being.

* I did not implement the icon button badge yet because I'm thinking the badge might be more versatile as a wrapper component like this. What do you think?

I 100% think it should be wrapped as the badge will be used in more places than the button. I think that's a good model to follow.

File: tailwind.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (iamjpg) [reply tailwind.config.js]: Ya, I should have pointed that out. This is why there is a `class="mds"` in the examples (https://codesandbox.io/s/moxi-design-system-esmodule-example-tw60n?file=/index.html:406-427)

**Review by iamjpg — APPROVED** [review]
Approving these changes. Looks awesome.

**Pull request merged**